src/nupic/engine/Link.cpp

```diff
@@ -388,6 +388,17 @@ void Link::purgeBufferHead()
   srcBuffer_.pop_front();
 }
 
+
+template <typename ST, typename ABT>
+void copyArrayToArrayProto(ST src, ABT builder, size_t elementCount)
```

Comment:
Would be nice to put this outside `Link.cpp` fro reuse since other classes will probably use `ArrayProto`